src/main/java/org/opensearch/security/support/Base64CustomHelper.java

```diff
@@ -0,0 +1,225 @@
+ */
+public class Base64CustomHelper {
+
+    private enum CustomSerializationFormat {
+
+        WRITEABLE(1),
+        STREAMABLE(2),
+        GENERIC(3);
+
+        private final int id;
+
+        CustomSerializationFormat(int id) {
+            this.id = id;
+        }
+
+        static CustomSerializationFormat fromId(int id) {
+            switch (id) {
+                case 1:
+                    return WRITEABLE;
+                case 2:
+                    return STREAMABLE;
+                case 3:
+                    return GENERIC;
+                default:
+                    throw new IllegalArgumentException(String.format("%d is not a valid id", id));
+            }
+        }
+
+    }
+
+    private static final BiMap<Class<?>, Integer> writeableClassToIdMap = HashBiMap.create();
+    private static final StreamableRegistry streamableRegistry = StreamableRegistry.getInstance();
+
+    static {
+        registerAllWriteables();
+    }
+
+    protected static String serializeObject(final Serializable object) {
+
+        Preconditions.checkArgument(object != null, "object must not be null");
+        final BytesStreamOutput streamOutput = new SafeBytesStreamOutput(128);
+        Class<?> clazz = object.getClass();
+        try {
+            prohibitUnsafeClasses(clazz);
+            CustomSerializationFormat customSerializationFormat = getCustomSerializationMode(clazz);
+            switch (customSerializationFormat) {
+                case WRITEABLE:
+                    streamOutput.writeByte((byte) CustomSerializationFormat.WRITEABLE.id);
+                    streamOutput.writeByte((byte) getWriteableClassID(clazz).intValue());
+                    ((Writeable) object).writeTo(streamOutput);
+                    break;
+                case STREAMABLE:
+                    streamOutput.writeByte((byte) CustomSerializationFormat.STREAMABLE.id);
+                    streamableRegistry.writeTo(streamOutput, object);
+                    break;
+                case GENERIC:
+                    streamOutput.writeByte((byte) CustomSerializationFormat.GENERIC.id);
+                    streamOutput.writeGenericValue(object);
+                    break;
+                default:
+                    throw new IllegalArgumentException(
+                        String.format("Could not determine custom serialization mode for class %s", clazz.getName())
+                    );
+            }
+        } catch (final Exception e) {
+            throw new OpenSearchException("Instance {} of class {} is not serializable", e, object, object.getClass());
+        }
+        final byte[] bytes = streamOutput.bytes().toBytesRef().bytes;
+        streamOutput.close();
+        return BaseEncoding.base64().encode(bytes);
+    }
+
+    protected static Serializable deserializeObject(final String string) {
+
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(string), "string must not be null or empty");
```

Comment:
Nit: Standardize message as "object must not be null or empty"

Reply:
Ack

Reply:
Done.